### PR TITLE
Add Google Analytics 4 Support

### PR DESCRIPTION
**Pull request by joshpuetz**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adds support for Google Analytics v4 to Forem.

Google recently accouncend they would be [discontinuing support for Google Analytics v3](https://support.google.com/analytics/answer/11583528) (sometimes called Universal Analytics) in July 2023. It's a long way away, but we decided to take on this work now for two reasons:

1) Google Analytics version 4 does not share data with version 3, so it's in our best interested to start gathering data now before we have to rely upon it soley

2) We originally thought the deadline was July 2022 😆 

This PR adds support for Google Analytics v4 to Forem. It's works just the same was as Google Analytics v3 support: a Forem creator/admin entered their Google Analytics ID in the Admin console, and tracking start starts flowing to Google. V3 is still supported, and Google recommends sites run both systems in parallel for a while to build up data and migrate their views.

## Related Tickets & Documents
- Closes https://github.com/forem/forem/issues/17979

## QA Instructions, Screenshots, Recordings

Admin Screen:
CleanShot 2022-07-14 at 14 49 37@2x

Data streaming from my local machine to Dev's GA4 account:
CleanShot 2022-07-14 at 14 50 37@2x

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [x] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

Dev's GA4 ID will need to be entered in the Admin console once this goes live

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/assets/javascripts/initializers/initializeBaseTracking.js.erb [hunk 1]
```diff
@@ -25,9 +31,48 @@ function initializeBaseTracking() {
     }, 25);
     eventListening();
   }
-  trackCustomImpressions();
 }
 
+function trackGoogleAnalytics4() {
```
Comment (joshpuetz) [inline app/assets/javascripts/initializers/initializeBaseTracking.js.erb:36]: SURPRISE! Google analytics v4 is a completely different script (gtag.js versus analytics.js) with different syntax. The upside is we can use both at the same time, but the downside is that you'll see I had to copy a lot of existing GAv3 code. 

This method is based on a great comment at ttps://stackoverflow.com/questions/22716542/google-analytics-code-explanation that breaks down how the old GA v3 async script loading work (Google doesn't provide such a script for GA4, they just expect you to hardcode it in)

File: app/assets/javascripts/initializers/initializeBaseTracking.js.erb [hunk 2]
```diff
@@ -25,9 +31,48 @@ function initializeBaseTracking() {
...
+
+        gtag('js', new Date());
+        gtag('config', ga4MeasurementCode, { 'anonymize_ip': true });
+        clearInterval(waitingOnGA4);
+      }
+      if (wait > 85) {
+        clearInterval(waitingOnGA4);
+        <%# The gem we're using server-side (Staccato) is not yet compatible with the Google Analytics 4 tracking code.
+        More details: https://github.com/tpitale/staccato/issues/97 %>
+        <%# fallbackActivityRecording(); %>
```
Comment (joshpuetz) [inline app/assets/javascripts/initializers/initializeBaseTracking.js.erb:68]: Definitely curious what @jeremyf thinks about this!

File: app/assets/javascripts/initializers/initializeBaseTracking.js.erb [hunk 3]
```diff
@@ -69,6 +114,15 @@ function eventListening(){
   }
 }
 
+function ga4eventListening(){
+  var registerNowButt = document.getElementById("cta-comment-register-now-link");
```
Comment (joshpuetz) [inline app/assets/javascripts/initializers/initializeBaseTracking.js.erb:118]: 🍑

File: config/locales/lib/en.yml [hunk 4]
```diff
@@ -108,7 +108,9 @@ en:
           favicon:
             description: Used as the site favicon
           ga_tracking:
-            description: Google Analytics Tracking ID, e.g. UA-71991000-1
+            description: Google Universal Analytics Tracking ID, e.g. UA-XXXXXXXX-1
```
Comment (joshpuetz) [inline config/locales/lib/en.yml:111]: Whoa: Dev's GA tracking ID was in the docs and showing up on all Forem's admin pages! 😱

**Review by jeremyf — APPROVED** [review]

File: app/assets/javascripts/initializers/initializeBaseTracking.js.erb (hunk 2 above)
Comment (joshpuetz) [reply app/assets/javascripts/initializers/initializeBaseTracking.js.erb:68]: Mostly I was wondering about the use of 'fallback' tracking. I think erroneously attributed the method to you based on some git history!

**Pull request merged**